server/http.go

```diff
@@ -64,138 +68,70 @@ const (
-
-	// Incoming webhook
-	case routeIncomingWebhook:
-		return p.httpWebhook(w, r, callbackInstanceID)
+	p.router.HandleFunc(routeACUserRedirectWithToken, p.handleResponseWithCallbackInstance(p.httpACUserRedirect)).Methods(http.MethodPost)
+	p.router.HandleFunc(routeACUserConfirm, p.handleResponseWithCallbackInstance(p.httpACUserInteractive)).Methods(http.MethodPost)
+	p.router.HandleFunc(routeACUserConnected, p.handleResponseWithCallbackInstance(p.httpACUserInteractive)).Methods(http.MethodPost)
+	p.router.HandleFunc(routeACUserDisconnected, p.handleResponseWithCallbackInstance(p.httpACUserInteractive)).Methods(http.MethodPost)
 
 	// Oauth1 (Jira Server)
-	case routeOAuth1Complete:
-		return p.httpOAuth1aComplete(w, r, callbackInstanceID)
-	case routeUserDisconnect:
-		return p.httpOAuth1aDisconnect(w, r, callbackInstanceID)
+	p.router.HandleFunc(routeOAuth1Complete, p.handleResponseWithCallbackInstance(p.httpOAuth1aComplete)).Methods(http.MethodPost)
+	p.router.HandleFunc(routeUserDisconnect, p.handleResponseWithCallbackInstance(p.httpOAuth1aDisconnect)).Methods(http.MethodPost)
 
 	// User connect/disconnect links
-	case routeUserConnect:
-		return p.httpUserConnect(w, r, callbackInstanceID)
-	case routeUserStart:
-		return p.httpUserStart(w, r, callbackInstanceID)
-	case routeAPIUserDisconnect:
-		return p.httpUserDisconnect(w, r)
+	p.router.HandleFunc(routeUserConnect, p.handleResponseWithCallbackInstance(p.httpUserConnect)).Methods(http.MethodPost)
+	p.router.HandleFunc(routeUserStart, p.handleResponseWithCallbackInstance(p.httpUserStart)).Methods(http.MethodGet)
+	p.router.HandleFunc(routeAPIUserDisconnect, p.handleResponse(p.httpUserDisconnect)).Methods(http.MethodPost)
 
 	// Firehose webhook setup for channel subscriptions
-	case routeAPISubscribeWebhook:
-		return p.httpSubscribeWebhook(w, r, callbackInstanceID)
-	}
+	apiRouter.HandleFunc(routeAPISubscribeWebhook, p.handleResponseWithCallbackInstance(p.httpSubscribeWebhook)).Methods(http.MethodPost)
 
-	if strings.HasPrefix(path, routeAPISubscriptionsChannel) {
-		return p.httpChannelSubscriptions(w, r)
-	}
+	// Channel Subscriptions
+	apiRouter.HandleFunc(routeAPISubscriptionsChannelWithID, p.handleResponse(p.httpChannelSubscriptions)).Methods(http.MethodGet)
+	apiRouter.HandleFunc(routeAPISubscriptionsChannel, p.handleResponse(p.httpChannelSubscriptions)).Methods(http.MethodPost)
+	apiRouter.HandleFunc(routeAPISubscriptionsChannel, p.handleResponse(p.httpChannelSubscriptions)).Methods(http.MethodPut)
+	apiRouter.HandleFunc(routeAPISubscriptionsChannelWithID, p.handleResponse(p.httpChannelSubscriptions)).Methods(http.MethodDelete)
+}
 
-	p.gorillaRouter.ServeHTTP(w, r)
-	return 0, nil
+func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	if strings.HasPrefix(r.URL.Path, "/plugins/servlet") {
+		body, _ := httputil.DumpRequest(r, true)
+		p.errorf("Received an unknown request as a Jira plugin:\n%s", string(body))
+	}
```

Comment:
I don't think this warrants logging as an error. Maybe a debug log if we do want to log. I know this code was already doing this before, but we might as well improve this here. Plus logging that body is going to be pretty big.

I honestly think we don't need to log this, but if we do, I think it should be debug log, and we should only log the URL instead of the body.

Reply:
Thanks, removed this code